### PR TITLE
feat(LockedField): add function to release lock for an asset

### PR DESCRIPTION
**Pull request by stonebuzz**
Add function to release all lock for an asset

| Q             | A
| ------------- | ---
| Bug fix?      | yes/no
| New feature?  | yes/no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

### CODE REVIEW

**Review by orthagh — APPROVED** [review]

File: src/Lockedfield.php [hunk 1]
```diff
@@ -284,6 +284,27 @@ public function showForm($ID, array $options = [])
...
+     *
+     * @param string $itemtype asset type
+     * @param int    $items_id asset ID
+     * @return boolean
+     * @since 10.0.4
+     */
+    public static function releaseLockForAsset($itemtype, $items_id)
+    {
+        global $DB;
+        return $DB->delete(
```
Comment (cedric-anne) [inline src/Lockedfield.php:298]: Using `$DB->delete()` directly will prevent usage of all `CommonDBTM` pre/post purge methods and will also prevent plugins to use pre/post purge hooks.
You should use `Lockedfield::delete()` method instead.

**Comment by trasher** [comment]
While I'm not against this change; 2 plugins will require a minor version of GLPI because of a quite simple change. See comments on inventory plugin; I'm not sure current PR is really needed, or at least method presence should be tested and a fallback on a deleteByCrit should be done.

File: src/Lockedfield.php [hunk 2]
```diff
@@ -34,6 +34,7 @@
  */
 
 use Glpi\Application\View\TemplateRenderer;
+use Sabre\DAV\Exception\Locked;
```
Comment (cedric-anne) [inline src/Lockedfield.php:37]:
```suggestion
```

File: src/Lockedfield.php [hunk 3]
```diff
@@ -294,14 +295,13 @@ public function showForm($ID, array $options = [])
      */
     public static function releaseLockForAsset($itemtype, $items_id)
     {
-        global $DB;
-        return $DB->delete(
-            Lockedfield::getTable(),
+        $lf = new Lockedfield();
+        $lf->deleteByCriteria([
```
Comment (cedric-anne) [inline src/Lockedfield.php:299]: There is no need to have an array in an array.

**Review by cedric-anne — CHANGES_REQUESTED** [review]

File: src/Lockedfield.php (hunk 1 above)
Comment (stonebuzz) [reply src/Lockedfield.php]: fixed

File: src/Lockedfield.php (hunk 3 above)
Comment (stonebuzz) [reply src/Lockedfield.php:298]: fixed

File: src/Lockedfield.php [hunk 4]
```diff
@@ -284,6 +284,24 @@ public function showForm($ID, array $options = [])
...
 
+    /**
+     * Clean locks for an asset
+     *
+     * @param string $itemtype asset type
+     * @param int    $items_id asset ID
+     * @return boolean
+     * @since 10.0.4
+     */
+    public static function releaseLockForAsset($itemtype, $items_id)
```
Comment (cedric-anne) [inline src/Lockedfield.php:295]:
```suggestion
    public static function releaseLockForAsset(string $itemtype, int $items_id): bool
```

File: src/Lockedfield.php [hunk 5]
```diff
@@ -284,6 +284,24 @@ public function showForm($ID, array $options = [])
...
+    {
+        $lf = new Lockedfield();
+        $lf->deleteByCriteria([
```
Comment (cedric-anne) [inline src/Lockedfield.php:298]:
```suggestion
        return $lf->deleteByCriteria([
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

**Comment by cedric-anne** [comment]
PR is not usefull anymore.

**Pull request closed without merging**